change replicas to instances

tiansuo114 opened the pull request:
change the replicas keyword in topoloy.yaml to instances
but also keep the old version,old version topoloy.yaml still can use

other:
Change the replicas keyword displayed after executing the curveadm status command to instances

将 topoloy.yaml 中的replicas关键字更改为instances
但也保留旧版本，旧版本的topoloy.yaml关键字仍然可以使用

其他：
将执行curveadm status命令后显示的replicas关键字改为instances

tiansuo114 commented:
Sorry I made a terrible mistake in old pull request(change the replicas keyword to service_num #294)😭😭
I have open a new pull request to restart that

Wine93 commented:
Hi @tiansuo114, have you tested it?

Wine93 commented:
@tiansuo114, you should also handle some related variables, like `service_replicas_sequence` and `service_replica_sequence`.

tiansuo114 commented:
> @tiansuo114 You should also handle some related variables, like `service_replicas_sequence` and `service_replica_sequence`.

my mistake😭i will change it

tiansuo114 commented:
> @tiansuo114 You should also handle some related variables, like `service_replicas_sequence` and `service_replica_sequence`.

I encountered some problems when changing the fields in LogicalPool in curveadm/internal/configure/pool.go. After I deleted the original replicas field and changed it to instances and compiled and ran it, I encountered a cpp file error, This file comes from curve/tools/curvefsTool.cpp. I don't know if I should change the cpp file or how to deal with this error. I chose to temporarily keep the replicas attribute to solve this problem. For this, I would like to ask for your opinion

![7c2b62516c32d3352e5d18012a1c979](https://github.com/opencurve/curveadm/assets/103884386/6eb37f86-cc1b-4b25-ae0c-ffa100a63974)

tiansuo114 commented:
> @tiansuo114 You should also handle some related variables, like `service_replicas_sequence` and `service_replica_sequence`.

I encountered some problems when changing the fields in LogicalPool in curveadm/internal/configure/pool.go. After I deleted the original replicas field and changed it to instances and compiled and ran it, I encountered a cpp file error, This file comes from curve/tools/curvefsTool.cpp. I don't know if I should change the cpp file or how to deal with this error. I chose to temporarily keep the replicas attribute to solve this problem. For this, I would like to ask for your opinion

![微信图片_20230830144028](https://github.com/opencurve/curveadm/assets/103884386/3983c018-0cb8-48c1-9fb5-33ba45ba4286)

tiansuo114 commented:
> @tiansuo114 You should also handle some related variables, like `service_replicas_sequence` and `service_replica_sequence`.

I encountered some problems when changing the fields in LogicalPool in curveadm/internal/configure/pool.go. After I deleted the original replicas field and changed it to instances and compiled and ran it, I encountered a cpp file error, This file comes from curve/tools/curvefsTool.cpp. I don't know if I should change the cpp file or how to deal with this error. I chose to temporarily keep the replicas attribute to solve this problem. For this, I would like to ask for your opinion

The following is the error message I encountered

Error-Code:410019
Error-Description: create logical pool failed
Error-Clue:
E 2023-08-30T14:17:28.23610+0800   134 curvefsTool.cpp:385] logicalpool replicasnum must be uin
E 2023-08-30T14:17:2823646+0800  134 curvefsTool.cpp:132] init logical pool data fail.
E 2023-08-30T14:17:28.023653+0800  134 curvefsTool.cpp:1109] exec fail，ret = -1
How to Solve:
* Website: https://github.com/opencurve/curveadm/wiki/errno4#410019
Log: /home/curve/.curveadm/logs/curveadm-2023-08-30_14-15-57.log
WeChat: opencurve_bot

tiansuo114 commented:
> > @tiansuo114 You should also handle some related variables, like `service_replicas_sequence` and `service_replica_sequence`.
> 
> I encountered some problems when changing the fields in LogicalPool in curveadm/internal/configure/pool.go. After I deleted the original replicas field and changed it to instances and compiled and ran it, I encountered a cpp file error, This file comes from curve/tools/curvefsTool.cpp. I don't know if I should change the cpp file or how to deal with this error. I chose to temporarily keep the replicas attribute to solve this problem. For this, I would like to ask for your opinion
> 
> The following is the error message I encountered
> 
> Error-Code:410019 Error-Description: create logical pool failed Error-Clue: E 2023-08-30T14:17:28.23610+0800 134 curvefsTool.cpp:385] logicalpool replicasnum must be uin E 2023-08-30T14:17:2823646+0800 134 curvefsTool.cpp:132] init logical pool data fail. E 2023-08-30T14:17:28.023653+0800 134 curvefsTool.cpp:1109] exec fail，ret = -1 How to Solve:
> 
> * Website: https://github.com/opencurve/curveadm/wiki/errno4#410019
>   Log: /home/curve/.curveadm/logs/curveadm-2023-08-30_14-15-57.log
>   WeChat: opencurve_bot

By the way, I don’t know if the changes in the source code will affect the image file deployed in curveadm (I guess the error of the cpp file in this file is due to a problem with the execution of the image file in docker), I hope Can you give suggestions for modifications?

Wine93 commented:
> > > @tiansuo114 You should also handle some related variables, like `service_replicas_sequence` and `service_replica_sequence`.
> > 
> > 
> > I encountered some problems when changing the fields in LogicalPool in curveadm/internal/configure/pool.go. After I deleted the original replicas field and changed it to instances and compiled and ran it, I encountered a cpp file error, This file comes from curve/tools/curvefsTool.cpp. I don't know if I should change the cpp file or how to deal with this error. I chose to temporarily keep the replicas attribute to solve this problem. For this, I would like to ask for your opinion
> > The following is the error message I encountered
> > Error-Code:410019 Error-Description: create logical pool failed Error-Clue: E 2023-08-30T14:17:28.23610+0800 134 curvefsTool.cpp:385] logicalpool replicasnum must be uin E 2023-08-30T14:17:2823646+0800 134 curvefsTool.cpp:132] init logical pool data fail. E 2023-08-30T14:17:28.023653+0800 134 curvefsTool.cpp:1109] exec fail，ret = -1 How to Solve:
> > 
> > * Website: https://github.com/opencurve/curveadm/wiki/errno4#410019
> >   Log: /home/curve/.curveadm/logs/curveadm-2023-08-30_14-15-57.log
> >   WeChat: opencurve_bot
> 
> By the way, I don’t know if the changes in the source code will affect the image file deployed in curveadm (I guess the error of the cpp file in this file is due to a problem with the execution of the image file in docker), I hope Can you give suggestions for modifications?

Please paste your toplogy.yaml.

Wine93 commented on internal/configure/pool.go:
```diff
@@ -96,7 +97,7 @@ type (
  *     - name: pool1
  *       physicalpool: pool1
- *       replicasnum: 3
+ *       instancessum: 3
```
Don't change it, the `replicasnum` represents the number of copies of the data, not the number of services, this is why we want to need this PR :)

Wine93 commented on internal/configure/pool.go:
```diff
@@ -119,7 +120,7 @@ type (
  *   pools:
  *     - name: pool1
- *       replicasnum: 3
+ *		 instancessum: 3
```
ditto.

Wine93 commented on internal/configure/pool.go:
```diff
@@ -198,17 +199,18 @@ func createLogicalPool(dcs []*topology.DeployConfig, logicalPool string, poolset
+		Name:      logicalPool,
+		Copysets:  copysets,
+		Zones:     zones,
+		Instances: DEFAULT_INSTANCES_PER_COPYSET,
```
You can delete it.

tiansuo114 commented:
Ok, I corrected my misunderstanding. After testing, the final version of the code can complete the deployment of the curve_bs cluster. I uploaded the deployment file that I executed in my test environment. If you need it, you can find it here download:
https://github.com/tiansuo114/go-tools/tree/main/curveadm_configs

My test is still not perfect, if you find any other mistakes, please ask me to modify the code again😭😭

Wine93 commented on cli/command/status.go:
```diff
@@ -73,7 +73,7 @@ func NewStatusCommand(curveadm *cli.CurveAdm) *cobra.Command {
 	flags.StringVar(&options.host, "host", "*", "Specify service host")
 	flags.BoolVarP(&options.verbose, "verbose", "v", false, "Verbose output for status")
-	flags.BoolVarP(&options.showReplicas, "show-replicas", "s", false, "Display service replicas")
+	flags.BoolVarP(&options.showInstances, "show-instances", "s", false, "Display service num")
```
```suggestion
flags.BoolVarP(&options.showInstances, "show-instances", "s", false, "Display service instances")
```

Wine93 reviewed (approved):
LGTM!
